### PR TITLE
[in-app-purchases] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1805

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `InAppPurchases` API - https://docs.expo.io/versions/latest/sdk/in-app-purchases - with minor tweaks and additions to the package source code.

To full support the migration to automated docs extraction I had to introduce few tweaks to the code:
* new `IAPPurchaseHistoryOptions` which describes the `getPurchaseHistoryAsync` parameter (but I'm not sure about the name),
* replacement of `number` types in few places with correct enum types, (ex. `InAppPurchase.purchaseState` or `getBillingResponseCodeAsync` returned type)
* simplification of `IAPQueryResponse` type,
* export of `QueryResult` type.

I have also placed the @default and @example values in the new lines (to increase the redability), earlier they were inline with the description.

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

Screenshot 2021-08-30 at 17 40 09

Screenshot 2021-08-31 at 17 25 29

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-in-app-purchases/src/InAppPurchases.ts [hunk 1]
```diff
@@ -34,6 +38,13 @@ const eventEmitter = new EventEmitter(ExpoInAppPurchases);
 let connected = false;
 let purchaseUpdatedSubscription: Subscription;
 
+// @needsAudit
+/**
+ * Connects to the app store and performs all of the necessary initialization to prepare the module
+ * to accept payments.
+ * This method must be called before anything else, otherwise an error will be thrown.
+ * @return Returns a Promise that fulfils when connection is established.
```
Comment (ide) [inline packages/expo-in-app-purchases/src/InAppPurchases.ts:46]:
```suggestion
 * @return Returns a Promise that fulfills when connection is established.
```

File: packages/expo-in-app-purchases/src/InAppPurchases.ts [hunk 2]
```diff
@@ -53,10 +105,25 @@ export async function getProductsAsync(
...
+ * restore completed transactions, therefore the `purchaseTime` and `orderId` may be inaccurate if
+ * it's a restored purchase. If you need the original transaction's information you can use
+ * `originalPurchaseTime` and `originalOrderId`, but those will be 0 and an empty string
+ * respectively if it is the original transaction.
+ *
+ * You should not call this method on launch because restoring purchases on iOS prompts for the
+ * user’s App Store credentials, which could interrupt the flow of your app.
+ *
+ * @param options An optional `PurchaseHistoryOptions` object.
+ * @return Returns a `Promise` that fulfils with an `IAPQueryResponse` that contains an array of
```
Comment (ide) [inline packages/expo-in-app-purchases/src/InAppPurchases.ts:122]:
```suggestion
 * @return Returns a `Promise` that fulfills with an `IAPQueryResponse` that contains an array of
```
etc..

**Comment by ide** [comment]
Looked very briefly, left a quick comment.

**Review by bbarthec — APPROVED** [review]
Looking fine to me 👍

**Review by tsapeta — APPROVED** [review]

**Pull request merged**